src/esp/core/Configuration.cpp

```diff
@@ -287,25 +306,53 @@ int Configuration::loadFromJson(const io::JsonGenericValue& jsonObj) {
       set(key, obj.GetString());
     } else if (obj.IsBool()) {
       set(key, obj.GetBool());
-    } else if (obj.IsArray() && obj.Size() > 0 && obj[0].IsNumber()) {
-      // numeric vector or quaternion
-      if (obj.Size() == 3) {
-        Magnum::Vector3 val{};
-        if (io::fromJsonValue(obj, val)) {
-          set(key, val);
+    } else if (obj.IsArray() && obj.Size() > 0) {
+      // non-empty array of values
+      if (obj[0].IsNumber()) {
+        // numeric vector or quaternion
+        if (obj.Size() == 3) {
+          Magnum::Vector3 val{};
+          if (io::fromJsonValue(obj, val)) {
+            set(key, val);
+          }
+        } else if (obj.Size() == 4) {
+          // assume is quaternion
+          Magnum::Quaternion val{};
+          if (io::fromJsonValue(obj, val)) {
+            set(key, val);
+          }
+        } else {
+          // decrement count for key:obj due to not being handled vector
+          --numConfigSettings;
+          // TODO support numeric array in JSON
+          ESP_WARNING()
+              << "Config cell in JSON document contains key" << key
+              << "referencing an unsupported numeric array of length :"
+              << obj.Size() << "so skipping.";
         }
-      } else if (obj.Size() == 4) {
-        // assume is quaternion
-        Magnum::Quaternion val{};
-        if (io::fromJsonValue(obj, val)) {
-          set(key, val);
+      } else if (obj[0].IsArray() && obj[0][0].IsNumber()) {
```

Comment:
I suppose this writes a JSON in a form of

```json
[[1, 0, 0],
 [0, 1, 0],
 [0, 0, 1]] 
```

right? I'd prefer an one-dimensional array instead, because:

- this is how glTF does it (in case you'd eventually want JSON compatibility to some degree)
- it's faster to parse, faster to read, faster to write, faster to validate (one just dumps array of 9 values, or checks that the numeric array has 9 values; which is especially convenient with the new [Utility::JsonToken::asFloatArray()](https://doc.magnum.graphics/corrade/classCorrade_1_1Utility_1_1JsonToken.html#ab0aeb26b5fe66652c95f6ab465718651) and [Utility::JsonWriter::writeArray()](https://doc.magnum.graphics/corrade/classCorrade_1_1Utility_1_1JsonWriter.html#a7f783366d73062d35e5a8059e8f01a0b) APIs, and the *also new* [Matrix::data()](https://doc.magnum.graphics/magnum/classMagnum_1_1Math_1_1RectangularMatrix.html#a18852de01a97e7f65398097520364288) API that returns a sized array, thus the copy from/to JSON can be a single `Utility::copy(json.asFloatArray(), matrix.data())` / `json.writeFloatArray(matrix.data())` call)
- there doesn't need to be any extra logic to flip from a column-major representation to a row-major representation, it's just a data copy

Reply:
My original thinking was that I wanted to make something that was a bit more human-readable, but I did notice the gltf standard (Which I am shooting for strict adherence to) so doing things this way makes sense to me.